junit-platform-engine/src/main/java/org/junit/platform/engine/support/hierarchical/ThrowableCollector.java

```diff
@@ -57,7 +67,11 @@ private void add(Throwable t) {
 		if (this.throwable == null) {
 			this.throwable = t;
 		}
-		else {
+		else if (this.throwable instanceof TestAbortedException && !(t instanceof TestAbortedException)) {
+			t.addSuppressed(this.throwable);
+			this.throwable = t;
+		}
+		else if (throwable != t) {
```

Comment:
What's the use case for this last `else if` block?

I'm not seeing it in my browser. 😉